executor/poolmgr/gp.go

```diff
@@ -574,9 +574,26 @@ func (gp *GenericPool) waitForReadyPod() error {
 		}
 
 		if time.Since(startTime) > gp.podReadyTimeout {
+			podList, err := gp.kubernetesClient.CoreV1().Pods(gp.namespace).List(metav1.ListOptions{
+				LabelSelector: labels.Set(
+					gp.deployment.Spec.Selector.MatchLabels).AsSelector().String(),
+			})
+			if err != nil {
+				log.Printf("Error getting pod list after timeout waiting for ready pod: %v", err)
+			}
+
+			// Since even single pod is not ready, choosing the first pod to inspect is a good approximation. In future this can be done better
+			pod := podList.Items[0]
+			var podErr string
+			for _, cStatus := range pod.Status.ContainerStatuses {
+				if cStatus.Ready != true {
+					podErr = cStatus.State.Waiting.Reason + cStatus.State.Waiting.Message
```

Comment:
It's better to use multierror package ("github.com/hashicorp/go-multierror") here to collect information for all unready contains. Also, it's better to have an error output format something like`fmt.Sprintf("Reason: %v, message: %v")`.

Reply:
@life1347 - I agree that this mechanism is not perfect - but collecting information for all containers might add too much information. For ex. if the pool size is 10 or 20 as an example - then we will have that many repeat messages. Since not a single pod was ready - it is a good approximation that all pods have the same error.